corgidrp/detector.py

```diff
@@ -520,7 +525,155 @@ def calc_sat_fwc(emgain_arr,fwcpp_arr,fwcem_arr,sat_thresh):
+        	std_resel (np.array):  Error of the flat field
+    """
+    n = im_size
+    full_qe=np.random.normal(1,.03,(n,n))
+    full_residuals = np.zeros((n,n))
+    if planet_rad==None:
+         planet_rad=54
+    
+    if rad_mask==None:
+         if band==1:
+              rad_mask==1.25
+         elif band==4:
+            rad_mask=1.75
+    
+    aperture = CircularAperture((np.ceil(rad_mask), np.ceil(rad_mask)), r=rad_mask)
+    mask= aperture.to_mask().data
+    rad = planet_rad
+    for i in np.arange(len(residual_images)):
+        nx = np.arange(0,residual_images[i].shape[1])
+        ny = np.arange(0,residual_images[i].shape[0])
+        nxx,nyy = np.meshgrid(nx,ny)
+        nrr = np.sqrt((nxx-rad-5)**2 + (nyy-rad-5)**2)
+        nrr_copy = nrr.copy()
+        nrr_copy[nrr<rad] = residual_images[i][nrr<rad] 
+        nrr_copy[nrr>=rad] = None 
+        ymin = int(cent[i][0])
+        ymax = int(cent[i][1])
+        xmin = int(cent[i][2])
+        xmax = int(cent[i][3])
+        
+        bool_innotzero = np.logical_and(nrr<rad,full_residuals[ymin:ymax,xmin:xmax]!=0)
+        bool_iniszero = np.logical_and(nrr<rad,full_residuals[ymin:ymax,xmin:xmax]==0)
+        bool_outisnotzero = np.logical_and(nrr>=rad,full_residuals[ymin:ymax,xmin:xmax]!=0)
+        
+        full_residuals[ymin:ymax,xmin:xmax][bool_innotzero] = (nrr_copy[bool_innotzero] + full_residuals[ymin:ymax,xmin:xmax][bool_innotzero])/2
+        full_residuals[ymin:ymax,xmin:xmax][bool_iniszero] = nrr_copy[bool_iniszero]
+        full_residuals[ymin:ymax,xmin:xmax][bool_outisnotzero] = full_residuals[ymin:ymax,xmin:xmax][bool_outisnotzero]
+        
+        
+        
+        full_qe_resel = ndimage.convolve(full_qe, mask)
+        full_residuals_resel = ndimage.convolve(full_residuals,mask)
+        std_resel = (((full_qe_resel - full_residuals_resel)/full_qe_resel))
+    
+    full_residuals[full_residuals==0] = None
+    nx = np.arange(0,std_resel.shape[1])
+    ny = np.arange(0,std_resel.shape[0])
+    nxx,nyy = np.meshgrid(ny,nx)
+    nrr = np.sqrt((nxx-n/2)**2 + (nyy-n/2)**2)
+    full_residuals[nrr>n_pix]= 1
+    std_resel[nrr>n_pix] = 0
+    full_qe[nrr>n_pix]= 1
+    full_qe=np.pad(full_qe, ((n_pad,n_pad),(n_pad,n_pad)), mode='constant',constant_values=(1))
+    full_residuals=np.pad(full_residuals, ((n_pad,n_pad),(n_pad,n_pad)), mode='constant',constant_values=(1))
+    std_resel=np.pad(std_resel, ((n_pad,n_pad),(n_pad,n_pad)), mode='constant',constant_values=(0))
+    
+    return (full_qe,full_residuals,std_resel)
+    
+    
+def create_onsky_flatfield(dataset, planet=None,band=None,up_radius=55,im_size=420,N=3,rad_mask=1.25,  planet_rad=54, n_pix=165, n_pad=302):
+    """Turn this dataset of raster scanned image frames of neptune or uranus into on-sky calibrated flat field 
+    
+        Args:
+            dataset (corgidrp.data.Dataset): a dataset of Image frames (L2a-level)
+            planet (str): neptune or uranus
```

Comment:
From my previous comment: If `planet` and `band` are not passed in, the code should be using the header keywords `TARGET` and `FILTER` to determine the planet and band. This makes it so that the user doesn't have to always manually change the `planet` and `band` when calling this function.